### PR TITLE
fix: resource scope

### PR DESCRIPTION
**Pull request by wansir**
Signed-off-by: hongming <talonwan@yunify.com>

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

Fix authorization logic, global resource priority over cluster resource.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for reviewers**:
```
```

**Additional documentation, usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/apiserver/apiserver.go [hunk 1]
```diff
@@ -238,6 +241,16 @@ func (s *APIServer) buildHandlerChain(stopCh <-chan struct{}) {
 	requestInfoResolver := &request.RequestInfoFactory{
 		APIPrefixes:          sets.NewString("api", "apis", "kapis", "kapi"),
 		GrouplessAPIPrefixes: sets.NewString("api", "kapi"),
+		GlobalResources: []schema.GroupResource{
+			iamv1alpha2.Resource(iamv1alpha2.ResourcesPluralUser),
+			iamv1alpha2.Resource(iamv1alpha2.ResourcesPluralGlobalRole),
+			iamv1alpha2.Resource(iamv1alpha2.ResourcesPluralGlobalRoleBinding),
+			tenantv1alpha1.Resource(tenantv1alpha1.ResourcePluralWorkspace),
+			tenantv1alpha2.Resource(tenantv1alpha1.ResourcePluralWorkspace),
```
Comment (zryfish) [inline pkg/apiserver/apiserver.go:249]: why there are two different workspaces?

Comment (wansir) [reply pkg/apiserver/apiserver.go:249]: `/kapis/tenant.kubesphere.io/v1alpha2/workspaces`, compatible with tenantv1alpha1 workspaces and tenantv1alpha2 workspacetemplates

**Pull request merged**